.circleci/config.yml

```diff
@@ -261,31 +78,74 @@ jobs:
             ./.ci/validation.sh git-check-pull-number
 
+  validation:
+    docker:
+      - image: &default-image checkstyle/jdk-11-groovy-git-mvn:11.0.13__3.0.9__2.25.1__3.6.3
+    parameters:
+      args:
+        type: string
+        default: ""
+        description: "argument to provide to validation.sh"
+    steps:
+      - run-script:
+          script-name: "./.ci/validation.sh"
+          args: << parameters.args >>
+
+  no-exception:
+    docker:
+      - image: *default-image
+    parameters:
+      args:
+        type: string
+        default: ""
+        description: "argument to provide to no-exception-test.sh"
+    steps:
+      - run-script:
+          script-name: "./.ci/no-exception-test.sh"
+          args: << parameters.args >>
+
+  no-error-hibernate-search:
+    docker:
+      - image: cimg/openjdk:18.0.1
+    steps:
+      - run-script:
+          script-name: "./.ci/validation.sh"
+          args: "no-error-hibernate-search"
+
 workflows:
-  version: 2
-  no-exception-tests:
+  test:
     jobs:
-      - no-exception-lucene-and-others-javadoc
-      - no-exception-cassandra-storm-tapestry-javadoc
-      - no-exception-hadoop-apache-groovy-scouter-javadoc
-      - no-exception-only-javadoc
-      - no-error-xwiki
-      - no-error-pmd
       - no-error-hibernate-search
-      - no-exception-struts
-      - no-exception-checkstyle-sevntu
-      - no-exception-checkstyle-sevntu-javadoc
-      - no-exception-guava
-      - no-exception-hibernate-orm
-      - no-exception-spotbugs
-      - no-exception-spoon
-      - no-exception-spring-framework
-      - no-exception-hbase
-      - no-exception-Pmd-elasticsearch-lombok-ast
-      - no-exception-alot-of-projects
-      - no-warning-imports-guava
-      - no-warning-imports-java-design-patterns
-      - sonarqube
+      - no-exception:
+          matrix:
+            parameters:
+              args: [
+                "no-exception-lucene-and-others-javadoc",
+                "no-exception-cassandra-storm-tapestry-javadoc",
+                "no-exception-hadoop-apache-groovy-scouter-javadoc",
+                "no-exception-only-javadoc",
+              ]
+      - validation:
+          matrix:
+            parameters:
+              args: [
+                "no-error-xwiki",
+                "no-error-pmd",
+                "no-exception-struts",
+                "no-exception-checkstyle-sevntu",
+                "no-exception-checkstyle-sevntu-javadoc",
+                "no-exception-guava",
+                "no-exception-hibernate-orm",
+                "no-exception-spotbugs",
+                "no-exception-spoon",
+                "no-exception-spring-framework",
+                "no-exception-hbase",
+                "no-exception-Pmd-elasticsearch-lombok-ast",
+                "no-exception-alot-of-projects",
+                "no-warning-imports-guava",
+                "no-warning-imports-java-design-patterns",
+                "sonarqube",
```

Comment:
We should probably consider moving these tasks to more aptly named files. I propose the following:
1. All `no-exception` should live in `no-exception-test.sh`, and only these tests
2. All `no-error` tests should live in a script called `no-error-test.sh`
3. All `no-warning` test should live in a script called `no-warning-test.sh`
4. Then we have `misc.sh for stuff like sonar.

Thoughts?

Reply:
1 and 2 ok.

3 and 4 are rare, better to keep in validation.sh

Reply:
Then perhaps 3 and 4 can be combined into misc?

Reply:
I don't like name - misc. What is a problem with 'validation' ?

Reply:
Agree with 1 and 2.

I don't have any issues with 3, but I don't know how many no-warnings we have. Also, how is that different than no-error? Warnings don't fail the build normally.

I agree with @romani on 4, better in validation.sh for now. validation is the misc of files.